File: src/leetcode/2026/02/solution628.java
import java.util.Arrays;

public class solution628 {
    public int maximumProduct(int[] nums) {
        int n = nums.length;
        Arrays.sort(nums);
        return Math.max(nums[0] * nums[1] * nums[n - 1], nums[n - 1] * nums[n - 2] * nums[n - 3]);
    }

    public static void main(String[] args) {
        solution628 solution = new solution628();

        int[] nums1 = { 1, 2, 3 };
        System.out.println(solution.maximumProduct(nums1)); // Should print 6

        int[] nums2 = { 1, 2, 3, 4 };
        System.out.println(solution.maximumProduct(nums2)); // Should print 24

        int[] nums3 = { -1, -2, -3 };
        System.out.println(solution.maximumProduct(nums3)); // Should print -6
    }
}

File: src/leetcode/2026/02/Solution628.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution628 {
public:
    int maximumProduct(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(), nums.end());
        return max(nums[0] * nums[1] * nums[n - 1], nums[n - 1] * nums[n - 2] * nums[n - 3]);
    }
};

int main() {
    Solution628 solution;

    vector<int> nums1 = {1, 2, 3};
    int result = solution.maximumProduct(nums1);
    cout << result << endl; // Output: 6

    vector<int> nums2 = {1, 2, 3, 4};
    result = solution.maximumProduct(nums2);
    cout << result << endl; // Output: 24

    vector<int> nums3 = {-1, -2, -3};
    result = solution.maximumProduct(nums3);
    cout << result << endl; // Output: -6

    return 0;
}